python/cudf/cudf/core/column/string.py

```diff
@@ -99,22 +99,31 @@ def _return_or_inplace(self, new_col, **kwargs):
         Returns an object of the type of the column owner or updates the column
         of the owner (Series or Index) to mimic an inplace operation
         """
-        from cudf import Series
+        from cudf import Series, DataFrame, MultiIndex
         from cudf.core.index import Index, as_index
 
         inplace = kwargs.get("inplace", False)
 
         if inplace:
-            self._column._mimic_inplace(new_col, inplace=True)
+            self._parent._mimic_inplace(new_col, inplace=True)
         else:
-            if isinstance(self._parent, Series):
+            expand = kwargs.get("expand", False)
+            if expand or isinstance(self._parent, (DataFrame, MultiIndex)):
+                return self._parent._constructor_expanddim(
+                    {index: value for index, value in enumerate(new_col)},
```

Comment:
May want to rename new_col here to indicate that we actually got a Table instead of a Column.

Reply:
@kkraus14, introduced a table var for it 👍